java/arcs/core/allocator/Allocator.kt

```diff
@@ -58,7 +51,7 @@ fun fieldAsStrings(entity: RawEntity, field: String): List<String> {
  */
 class Allocator private constructor(
     private val hostRegistry: HostRegistry,
-    private val collection: CollectionHandle<RawEntity>
+    private val collection: ReadWriteCollectionHandleAdapter<EntityBase>
```

Comment:
I wonder if this should be a general `PlanSerializer` interface, with a default implementation using handles?